File: solutions/go/proverb/1/proverb.go
// Package proverb provides a method to build a proverb from a given List
package proverb

import "fmt"

// Proverb is created based on a given list of words
func Proverb(rhyme []string) (text []string) {
	if len(rhyme) == 0 {
		return
	}
	for index, word := range rhyme {
		if index == len(rhyme)-1 {
			break
		}
		text = append(text, fmt.Sprintf("For want of a %s the %s was lost.", word, rhyme[index+1]))
	}
	text = append(text, fmt.Sprintf("And all for the want of a %s.", rhyme[0]))
	return
}

File: solutions/python/proverb/2/proverb.py
def proverb(*words, qualifier):
    lines = [
        f"For want of a {words[index]} the {words[index + 1]} was lost."
        for index in range(len(words) - 1)
    ]
    if words:
        lines.append(
            f"And all for the want of a {qualifier + ' ' if qualifier else ''}{words[0]}."
        )
    return lines
